### ISSUE TITLE
[Bug Report] Visual Studio Code consuming a lot of RAM

### ISSUE THREAD

**Issue by glaucia86** [microsoft/vscode#93276]
- Visual Studio Code Version: 1.43.0
- **Commit:** 78a4c91400152c0f27ba4d363eb56d2835f9903a
- **Date:** 2020-03-09T19:44:52.965Z
- **Electron:** 7.1.11
- **Chrome:** 78.0.3904.130
- **Node.js:** 12.8.1
- **V8:** 7.8.279.23-electron.0
- **OS:** Linux x64 4.4.0-93-generic
- **Visual Studio Code Extensions:** EditorConfig (v0.14.4) & LiveShare (v1.0.1742)

Steps to Reproduce:

In a thread on twitter, a Brazilian JavaScript Developer (**[https://twitter.com/fdaciuk/status/1239117094282833923](https://twitter.com/fdaciuk/status/1239117094282833923)**), reported problems with his Visual Studio Code for consuming a lot of RAM.
I saw the thread and talked to him via DM on Twitter to understand his machine and what extensions he had on his machine.

The developer was browsing through some files and left Visual Studio Code open for a few hours. So, the developer realized that the memory was too high in his machine.

When he went to see which softwares was open he realized his Visual Studio Code was open. Upon closing, immediately released 4Gb of memory from his machine.

Does this issue occur when all extensions are disabled?: Yes/No -->> **No**

**Comment by mjbvz** [comment]
[These instructions](https://github.com/Microsoft/vscode/wiki/Performance-Issues) describe how to investigate performance information.

However if this only happens when extensions are enabled, then one of them is causing this

**Comment by dalisoft** [comment]
I today also got this problem.

There some setup on mac, should work to you too.

1. Run some mid-level Python project
2. Let it done analysis
3. Run file with 50+ LOC
4. Change font by CMD + (+/-)
5. Watch from monitor how RAM grows faster than speed of the light 😁

**Comment by mjbvz** [comment]
@Dalisoft Test with extensions disabled. If you still see the problem, open a separate issue

**Comment by linusbrolin** [comment]
I have the same problem.
The RAM usage can climb to several gigabytes!
I have turned off all extensions, but after reload the RAM usage is still ~1.5GB and the CPU fan spins up and down all the time (even when I'm not doing anything), like it's doing some heavy lifting.

Simple scrolling in the vscode editor is laggy.
Ctrl-clicking on a function call or variable to get to its definition (in the same file) can take forever.

This is in a relatively large Angular/Node.js project, but it was not nearly this bad just a few days ago (I had no issue with scrolling, no issue with the cpu fan spinning up an down constantly, and (almost) no issue ctrl-clicking to get to definition in the same file).

So my guess is that it is something in the language service that is constantly running and scanning the entire codebase, and thus causing a worse and worse problem the larger the codebase becomes.
Could it be the semantic highlighting? (even though I've disabled it?)
